### PR TITLE
Simplify guide and make it more prescriptive

### PR DESCRIPTION
**Pull request by caabernathy**
This is related to #8157. Updated the guide and made it simpler and more prescriptive.

**Testing**

- cd react-native
- npm install
- cd website
- npm install
- npm start
- Go to http://localhost:8079/react-native/docs/platform-specific-code.html
- Checked content

platform-specific-update

### CODE REVIEW

**Comment by facebook-github-bot-6** [comment]
By analyzing the blame information on this pull request, we identified @grabbou and @emilioicai to be potential reviewers.

File: docs/PlatformSpecificInformation.md [hunk 1]
```diff
@@ -7,34 +7,12 @@ permalink: docs/platform-specific-code.html
 next: native-modules-ios
 ---
 
-When building a cross-platform app, the need to write different code for different platforms may arise. This can always be achieved by organizing the various components in different folders:
+When building a cross-platform app, you'll want to re-use as much code as possible. Scenarios may arise where it makes sense for the code to be different, for example you may want to implement separate visual components for iOS and Android.
```
Comment (JoelMarcey) [inline docs/PlatformSpecificInformation.md]: nit: Maybe a `;` instead of a `,` here.

**Comment by JoelMarcey** [comment]
@caabernathy I had one nit. Not worth blocking. Let's shipit.

**Comment by JoelMarcey** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-8** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1772619919617936/int_phab) to review.

**Pull request closed without merging**